### DISCUSSION TITLE
FDRS hardware

### DISCUSSION THREAD

**Discussion by aviateur17** [timmbogner/Farm-Data-Relay-System#160] [Show and tell]
Examples of some hardware I use for FDRS

Wemos form factor ESP32 with Hallard Shield.  Can use ESP-NOW or WiFi and LoRa 
![PXL_20230407_152831340 MP](https://user-images.githubusercontent.com/61305397/230636196-34d31fa7-d462-4720-9674-62e98eff4c67.jpg)

Seeed Xiao ESP32-S3 with OLED and PIR
![PXL_20230407_151213117 MP](https://user-images.githubusercontent.com/61305397/230636333-ee8becd0-3db8-43b8-94c7-1680c32e0dfd.jpg)

Wemos form factor ESP32 with OLED and PIR and BMP-280 powered by 18650 battery
![PXL_20230407_151131117 MP](https://user-images.githubusercontent.com/61305397/230636363-426847a5-6442-4ae2-b7fc-267a067edff2.jpg)

**Comment by timmbogner** [comment]
I really like that Hallard module! Is your antenna just a length of solid copper wire?
Currently my setup is just the [ThingPulse ethernet gateway](https://thingpulse.com/product/espgateway-ethernet-esp32-wifi-ble-gateway-with-rj45-ethernet-connector/). It's configured to have a second, LoRa gateway connected by ESP-NOW, but I don't have any sensors that need it as of yet.
I recently implemented a solar node with an ESP8266 reporting at one minute intervals. Unfortunately I didn't include any voltage monitoring, so it's just fingers crossed that it'll last forever 😄
I'll post some pics when I get a chance.

**Reply by aviateur17** [reply]
Yep, just a length of copper wire approx 78mm long.
I'd love to see and hear about your solar node. I haven't played with powering anything with solar yet.

**Reply by bvwelch** [reply]
That hallard module looks very cool! Did you solder it yourself?  A quick google found this page: https://github.com/hallard/WeMos-Lora
thanks.

**Reply by aviateur17** [reply]
Yeah, just ordered the PCBs from the site linked in that github link and the LoRa chips from Amazon and the other components from Digikey.  Once they all came in I soldered them on the board.  Pretty difficult to solder that stuff without some good tools but it seems to work.  I think they are 0508 size components which are probably the smallest I can do by hand.

**Reply by bvwelch** [reply]
My smt skills are not so much, so I may just get one of these when adding lora to a esp32 board.

https://www.adafruit.com/product/3072

**Comment by timmbogner** [comment]
Here's my new solar node! It has proven itself in the last couple days, producing more voltage than consuming. It has a very small battery, very small solar panel, and is placed in partial shade, but the TTGO T-Display inside is reporting the temp, soil moisture, and battery level every minute.
![IMG_20230424_133917391_HDR](https://user-images.githubusercontent.com/64260873/234131916-64500628-dcdb-4bf8-a0c7-5926f62e7a97.jpg)
Battery level over 24hrs:
![chart](https://user-images.githubusercontent.com/64260873/234131929-7a5af5fa-c3cb-42b0-89ce-e4157c132d94.png)


- [The solar module](https://www.aliexpress.us/item/3256804393091454.html)
- [The solar panel](https://www.aliexpress.us/item/3256804497786259.html)

**Reply by bvwelch** [reply]
very cool!

**Reply by bvwelch** [reply]
Here is the panel I have. But I haven't tested it yet. 

https://www.amazon.com/gp/product/B074TYH68Z/

**Reply by timmbogner** [reply]
> Here is the panel I have. But I haven't tested it yet.

Coincidentally, I ordered this very similar one last night (the price also rose by $3 after I bought it):
https://www.amazon.com/dp/B081YXGT2M

**Reply by bvwelch** [reply]
Note to self- when I do test it, find the Swiss guy's video that explains how to modify a resistor to allow the proper charge rate between the panel and the controller.  I might be the one with the Indiana Jones clip.

**Reply by timmbogner** [reply]
A resistor on the MPPT module? Which controller are you using?

**Comment by bvwelch** [comment]
It isnt a real mppt module, see discussion at 6 or 7 minutes into the video

https://youtu.be/37kGva3NW8w

**Reply by timmbogner** [reply]
Ah, right! I forgot about that one.

**Comment by javrolijk** [comment]
Hi folks.
Just wanted to chime in. I ordered this Solar Power Management Module combo. Seems to integrate pretty much all that's needed for proper and safe charging of the battery and power supply. It's $14 bucks, I haven't had the chance to test it out yet.

https://www.amazon.com/dp/B07PBRK8KG

**Reply by timmbogner** [reply]
That's super-handy. I'm pleased to see that it is listed [cheaper when ordered from the manufacturer](https://www.waveshare.com/solar-power-manager.htm) too! I'm not sure what shipping is though...

That really is a nice all-in-one module. I have several recycled panels of different voltages, so having variable panel voltage would be nice. The 5V also makes it useful as a power bank. The obscure (to me) 14500 battery size is a bit awkward, but I guess they couldn't fit an 18650.

Waveshare also has a cool[ LoRa hat for the Pi Pico](https://www.waveshare.com/pico-lora-sx1262-868m.htm) that is FDRS compatible.

**Comment by timmbogner** [comment]
This gateway has now spent about 48 consecutive hours online running off of the sun! The voltage readings are a bit skewed, not sure if that's a math error or what, but the important thing is it stays alive.
- The battery was salvaged and split from a larger pack a couple years ago. I don't recall its specifications, but I think it's in the 3000mAh range.
- The panel was bought at a junk store years ago, and is rated 12V 5W.
- [Here's a link](https://www.aliexpress.us/item/2255800893485745.html) to the MPPT module.
- And of course, everyone loves [LilyGo's TTGO T-Display](https://www.lilygo.cc/products/lilygo%C2%AE-ttgo-t-display-1-14-inch-lcd-esp32-control-board)🙂 

Note: This gateway was not receiving any traffic. That's the next step!

![solar1](https://github.com/timmbogner/Farm-Data-Relay-System/assets/64260873/1cef8cb6-31ba-4dd9-bf42-2d31618ca3a1)
![solar3](https://github.com/timmbogner/Farm-Data-Relay-System/assets/64260873/79ea7cb8-6860-492e-a515-5724a592a41c)
![chartgtwy2](https://github.com/timmbogner/Farm-Data-Relay-System/assets/64260873/f2fdb6bf-04b9-45af-8e8d-3cfb2b1271aa)
